pkg/logs/logs.go

```diff
@@ -96,10 +84,15 @@ func Start() error {
 	atomic.StoreInt32(&isRunning, 1)
 	log.Info("logs-agent started")
 
-	// add the default sources
-	for _, source := range config.DefaultSources() {
-		sources.AddSource(source)
-	}
+	// add the default sources after the AutoConfig has been executed once because
+	// we don't want the container_collect_all to reap all containers ownership
+	go func() {
+		common.BlockUntilAutoConfigRanOnce(time.Millisecond * coreConfig.Datadog.GetDuration("ac_load_timeout"))
+		log.Debug("Adding DefaultSources to the Logs Agent")
+		for _, source := range config.DefaultSources() {
```

Comment:
Does it make sense to wait for _all_ default sources? (I think there can be 2 right now: "ContainerCollectAll" and the snmp trap source).

Typically I don't think we need to wait for the snmp trap source (I think waiting could actually make us miss some snmp traps while Autodiscovery finishes its first collection). To avoid this I'd be OK with separating the setup of the ContainerCollectAll source (async) from the snmp traps source (sequential in `Start`), wdyt?

Reply:
My last commits should have addressed all your feedback.